controllers/drplacementcontrol.go

```diff
@@ -1524,6 +1524,7 @@ func (d *DRPCInstance) generateVRG(dstCluster string, repState rmn.ReplicationSt
 			Namespace: d.vrgNamespace,
 			Annotations: map[string]string{
 				DestinationClusterAnnotationKey: dstCluster,
+				DoNotDeletePVCAnnotation:        d.instance.GetAnnotations()[DoNotDeletePVCAnnotation],
```

Comment:
Why do we need to annotate the drpc? This looks like attempt to keep backward compatibility, but I don't see why we need to keep the old behavior. This duplicate the testing effort - now we have to test 2 cases for every test case, one with the annotation, and one without.

We agreed that the old behavior - deleting the pvc when the primary vrg is deleted is a bug, and we want to own only the pvc on the secondary pvc.

This change requires manual work from the admin, either add this annotation to all drpcs, or worse, add it before disabling dr, and is applied to both primary secondary vrg (or maybe generateVRG is used only for the primary?).

Reply:
> Why do we need to annotate the drpc? This looks like attempt to keep backward compatibility, but I don't see why we need to keep the old behavior. This duplicate the testing effort - now we have to test 2 cases for every test case, one with the annotation, and one without.

Annotation is the best way to add support for disabling DR. There is no control over when to NOT own the PVC.

Reply:
> We agreed that the old behavior - deleting the pvc when the primary vrg is deleted is a bug, and we want to own only the pvc on the secondary pvc.

The old behavior was not a bug. It was a workaround.  Owning the PVC only when the VRG is secondary is not enough. As I said in my response above, we don't have control over when the PVC is deleted on failover/relocate/VRG Deletion.

Reply:
> This change requires manual work from the admin, either add this annotation to all drpcs, or worse, add it before disabling dr, and is applied to both primary secondary vrg (or maybe generateVRG is used only for the primary?).

Yes, we want that and @netzzer is fine with it.

Reply:
We do have enough context with RBD and we don't need to add anything to the drpc when we disable dr, so I think it should work for volsync case in the same way, but I don't understand the flows enough to tell.

If our issue is having more context when we delete the vrg, maybe we can do this:

1. Handle the update event when the drpc is deleted (old object does not have a delete timestamp, new object has a deleteTimestamp)

1. In the update event, annotate the vrg

   ```
   drplacementcontrol.ramendr.openshift.io/dr-disabled: "true"
   ```
   This annotation will propagate to the primary vrg (same way the do-not-delete annotation is)

1. Wait until the vrg is updated

1. In vrg delete flow, disown the pvc

With this the admin does not have do anything when drpc is deleted.

Even if we don't have time now to change the flow, a declarative annotation like "dr-disabled" is better then "do-not-delete" since we can extend the behavior later without adding another annotation or using "do-not-delete" annotation for unrelated behavior.

Reply:
> We do have enough context with RBD and we don't need to add anything to the drpc when we disable dr, so I think it should work for volsync case in the same way, but I don't understand the flows enough to tell.

The follow for RBD mirroring is different from VolSync Replication. In other words, RBD snapshots for mirroring works differently than VolSync taken snapshots, where one does not care about PVC states, whereas the other cares. Saying it differently, VolSync Replication snapshots are k8s managed VolumeSnapshots, vs. RBD snapshots which are Ceph snapshots.

Reply:
> Even if we don't have time now to change the flow, a declarative annotation like "dr-disabled" is better then "do-not-delete" since we can extend the behavior later without adding another annotation or using "do-not-delete" annotation for unrelated behavior.

The 'do-not-delete' annotation explicitly conveys (the user), when protecting my workload PVCs, regardless of any actions taken, in the end is to leave them alone and unchanged.

If a naming change were to be considered, a candidate could be `when-protecting-my-workload-do-not-mess-with-my-protected-resources-state.`
Remember, disabling DR is an afterthought.  All we are doing here is; As a user, if years from now I decide to disable DR for a specific workload, I should be able to do so with some help from me (the user).

I agree, although it would be more convenient if we could avoid doing that. However, the nature of our interactions with other components necessitates this and I can't see how we can avoid it.